src/error.rs

```diff
@@ -411,6 +412,11 @@ pub enum ErrorKind {
     #[error("{message}")]
     #[non_exhaustive]
     Transaction { message: String },
+
+    /// The server version does not support the operation.
+    #[error("The server version does not support a database operation: {message}")]
+    #[non_exhaustive]
+    ServerVersion { message: String },
```

Comment:
What do you think about naming this `IncompatibleServerVersion` instead for some added clarity? Or even `IncompatibleServer` in case there are more criteria in the future unrelated to server version that may cause an operation to fail?

Reply:
Good call, done.